### PR TITLE
Fix Vsphere rouding up of volume size

### PR DESCRIPTION
**Pull request by bertinatto**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Storage plugins treat numbers-only storage size as bytes, however, they round it up to the nearest allocation unit, which is vendor-dependant. The Vsphere plugin, on the other hand, rounds the value up to at least 1 KiB and dispatches the request to the server. Since Vsphere's minimum allocation size is 1 MiB, the request returns an error.

This PR changes the Vsphere plugin so it acts like the other storage plugins. Even though the rounding might not be ideal (doing something different from what was requested by the user), we should have consistency across the the storage plugins.

```release-note
NONE
```

/sig storage

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
@kubernetes/cloud-provider-vsphere-maintainers

**Comment by bertinatto** [comment]
CC @saad-ali 
for approval

File: test/e2e/storage/vsphere/vsphere_volume_disksize.go
Comment (bertinatto) [inline test/e2e/storage/vsphere/vsphere_volume_disksize.go]: Error validation in this package is done in many different ways, example:
```
Expect(err).To(HaveOccurred())
framework.ExpectNoError(err, fmt.Sprintf("Failed to create storage class with err: %v", err))
framework.ExpectNoError(err)
```
For consistency's sake, in this patch I only used `framework.ExpectNoError(err)` .

File: test/e2e/storage/vsphere/vsphere_volume_disksize.go
Comment (bertinatto) [inline test/e2e/storage/vsphere/vsphere_volume_disksize.go]: It seems we don't run these tests at all, however, I updated this one to make sure it works.

**Comment by gnufied** [comment]
looks good to me. But I want @divyenpatel @BaluDontu to chime in as well.

/approve

**Comment by frapposelli** [comment]
/lgtm
/priority important-soon

/hold

(holding until somebody from the storage team reviews)

**Comment by gnufied** [comment]
@frapposelli since I have already reviewed from storage team we can remove the hold. Or did you mean someone from vmware storage team?

**Comment by divyenpatel** [comment]
`RoundUpToMiBInt` looks good to me.

**Review by divyenpatel — APPROVED** [review]
/lgtm

**Pull request merged**
